### ISSUE TITLE
Difference in behavior based on OS 

### ISSUE THREAD

**Issue by Rob192** [mindee/doctr#385]
## 🐛 Bug

I ran into this and I wanted to share with you. Maybe you know if something can be done. There is a difference in the extracted information you get when you treat a document (in my case French ID cards in .jpg) with doctr running on Windows or running in a dockerised Linux environment. 

## To Reproduce

1. Extract each words from the result of DocumentFile.from_images(doc) on Windows and on a Docker
2. Compare the extracted box content and position

Windows 
              word     min_x     min_y  ...  page_id  document_name  label
0        FRANÇAISE  0.535156  0.092773  ...        0              1      O
1       REPUBLIQUE  0.097656  0.106445  ...        0              1      O
2      Nationalité  0.703125  0.177734  ...        0              1      O
3       D'IDENTITE  0.284180  0.184570  ...        0              1      O
4               No  0.394531  0.184570  ...        0              1      O
5        Française  0.807617  0.176758  ...        0              1      O
6   CARTENATIONALE  0.101562  0.185547  ...        0              1      O
8             Nom:  0.335938  0.249023  ...        0              1      O
11       Prénom(s)  0.340820  0.349609  ...        0              1      O
14         Né(elle  0.560547  0.435547  ...        0              1      O
16          Sexe:M  0.333984  0.434570  ...        0              1      O
17               à  0.346680  0.501953  ...        0              1      O

Docker 
              word     min_x     min_y  ...  page_id  document_name  label
0        FRANÇAISE  0.535156  0.092773  ...        0              1      O
1       REPUBLIQUE  0.097656  0.106445  ...        0              1      O
2      Nationalité  0.702148  0.177734  ...        0              1      O
3       D'IDENTITE  0.286133  0.183594  ...        0              1      O
4        Française  0.807617  0.177734  ...        0              1      O
5              Nos  0.395508  0.183594  ...        0              1      O
6   CARTENATIONALE  0.101562  0.185547  ...        0              1      O
8             Nom:  0.335938  0.249023  ...        0              1      O
11       Prénem(s)  0.341797  0.348633  ...        0              1      O
14            Sexe  0.339844  0.441406  ...        0              1      O
15         Né(e)le  0.560547  0.433594  ...        0              1      O
18               a  0.343750  0.500977  ...        0              1      O

## Expected behavior

Results should be the same

## Environment

Windows 
DocTR version: 0.3.0
TensorFlow version: 2.5.0
PyTorch version: N/A (torchvision N/A)
OpenCV version: 4.5.2
OS: Microsoft Windows 11 Pro
Python version: 3.8
Is CUDA available (TensorFlow): No
Is CUDA available (PyTorch): N/A
CUDA runtime version: 11.0.221
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\bin\cudnn_ops_train64_8.dll

Docker 
DocTR version: 0.3.0
TensorFlow version: 2.5.0
PyTorch version: N/A (torchvision N/A)
OpenCV version: 4.5.2
OS: Debian GNU/Linux 10 (buster)
Python version: 3.8
Is CUDA available (TensorFlow): No
Is CUDA available (PyTorch): N/A
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

**Comment by fg-mindee** [comment]
Hey @Rob192,

Thanks for reporting this! That's indeed a strange behaviour consider most of the word predictions are identical :thinking: 
It's gonna be tricky to find, but it might indicate that the segmentation --> box conversion, or the cropping functions are behaving differently depending on the OS.

If it was coming from the TF models themselves, none of the output would be the same. Would you mind checking the output of the detection predictor in both cases? :pray:  (or if you can share the image that you used)

**Comment by Rob192** [comment]
Hello @fg-mindee, 
I realized that the difference between the two ouput actually came from the preprocessing I am doing before feeding the image to Doctr. For my use case, I use `cv2.SIFT_create` + `cv2.FlannBasedMatcher` to match a reference CNI card at a reference position. The image resulting from this transformation is clearly different between the Docker and Windows for whatever reason. Therefore Doctr gives different results as well. If I bypass this preprocessing step, there is no difference between Doctr outputs on Windows or Linux. 
Sorry for the confusion, I think we can close this.

**Comment by fg-mindee** [comment]
Thanks for the clarification @Rob192 :pray:
